maat-court-data-api/src/test/java/gov/uk/courtdata/courtdataadapter/client/CourtDataAdapterClientIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+import gov.uk.courtdata.model.laastatus.LaaStatusUpdate;
+import gov.uk.courtdata.model.laastatus.RepOrderData;
+import gov.uk.courtdata.service.QueueMessageLogService;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import org.slf4j.LoggerFactory;
+
+@ExtendWith(MockitoExtension.class)
+class CourtDataAdapterClientIntegrationTest {
+
+    private static final Integer MAX_RETRIES = 3;
+    private static final Integer MIN_BACKOFF_PERIOD = 1;
+    private static final Double JITTER = 0.75;
+    private static final String LOCALHOST = "http://localhost:%s";
+    private static final String LAA_STATUS_URL = "cda-test/laaStatus";
+    public static MockWebServer mockCourtDataAdapterApi;
+
+    private String baseUrl;
+
+    @Mock
+    private CourtDataAdapterClientConfig courtDataAdapterClientConfig;
+    @Mock
+    private QueueMessageLogService queueMessageLogService;
+
+    private GsonBuilder gsonBuilder;
+    private CourtDataAdapterClient courtDataAdapterClient;
+    private WebClient webClient;
+
+    @Mock
+    private OAuth2AuthorizedClientManager authClientManager;
+
+    private ListAppender<ILoggingEvent> listAppender;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        gsonBuilder = new GsonBuilder();
+        mockCourtDataAdapterApi = new MockWebServer();
+        mockCourtDataAdapterApi.start();
+        baseUrl = String.format(LOCALHOST, mockCourtDataAdapterApi.getPort());
+        webClient = buildWebClient();
+
+        Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
+    }
+
+    private WebClient buildWebClient() {
+        CourtDataAdapterOAuth2ClientConfig courtDataAdapterOAuth2ClientConfig = new CourtDataAdapterOAuth2ClientConfig();
+        return courtDataAdapterOAuth2ClientConfig.webClient(baseUrl, authClientManager, MAX_RETRIES, MIN_BACKOFF_PERIOD, JITTER);
+    }
+
+    @Test
+    public void givenAValidLaaStatusObject_whenPostLaaStatusIsInvoked_andSomeErrorResponsesReceivedFromCDAAreUnderMaxRetries_thenTheRequestIsSentCorrectlyAndAnAcceptedResponseIsReceived() {
```

Comment:
Also SonarCloud would like the public modifiers removed from the tests!

Reply:
Done.